usaspending_api/awards/serializers_v2/serializers.py

```diff
@@ -346,6 +355,104 @@ class Meta:
         }
 
 
+class AwardIDVSerializerV2(LimitableSerializerV2):
+    latest_transaction_contract_data = serializers.SerializerMethodField('latest_transaction_func')
+    idv_dates = serializers.SerializerMethodField("idv_dates_func")
+    executive_details = serializers.SerializerMethodField("executive_details_func")
+    parent_generated_unique_award_id = serializers.SerializerMethodField('parent_unique_id_func')
+
+    def idv_dates_func(self, award):
+        transaction_data = self.latest_transaction_func(award)
+        return {
+            "start_date": award.period_of_performance_start_date,
+            "last_modified_date": award.last_modified_date,
+            "end_date": transaction_data["ordering_period_end_date"]
+        }
+
+    def latest_transaction_func(self, award):
+        return TransactionFPDSSerializerV2(award.latest_transaction.contract_data).data
+
+    def executive_details_func(self, award):
+        entity = LegalEntityOfficerPassThroughSerializerV2(award.recipient).data
+        response = []
+        if "officers" in entity and entity["officers"]:
+            for x in range(1, 6):
+                response.append({"name": entity["officers"]["officer_" + str(x) + "_name"],
+                                 "amount": entity["officers"]["officer_" + str(x) + "_amount"]})
+        return {"officers": response}
+
+    # todo: revert parent generated unique id back to its original purpose  -jl
+    def parent_unique_id_func(self, award):
+        parent_transaction = TransactionFPDS.objects.filter(agency_id=award.fpds_parent_agency_id,
+                                                            piid=award.parent_award_piid).values(
+            "agency_id",
+            "referenced_idv_agency_iden",
+            "piid",
+            "parent_award_id").first()
+
+        if parent_transaction:
+            parent_generated_unique_id = (
+                "CONT_AW_" +
+                (parent_transaction["agency_id"] if parent_transaction["agency_id"] else "-NONE-") +
+                "_" +
+                (parent_transaction["referenced_idv_agency_iden"] if parent_transaction[
+                    "referenced_idv_agency_iden"] else "-NONE-") +
+                "_" +
+                (parent_transaction["piid"] if parent_transaction["piid"] else "-NONE-") +
+                "_" +
+                (parent_transaction["parent_award_id"] if parent_transaction["parent_award_id"] else "-NONE-")
+            )
```

Comment:
Is there talk about changing this to just query for this field directly from the awards table once IDV work is done? I worry that generating this here adds unnecessary complexity and risk.